### ISSUE TITLE
useLazyQuery().load() called twice api

### ISSUE THREAD

**Issue by Dykmi** [Akryum/vue-apollo#1444]
when:
```
const lazyQuery = useLazyQuery(document, variables);
lazyQuery.load(document, variables);
```
goes:
graphql api called twice

when:
```
const lazyQuery = useLazyQuery(document, variables);
setTimeout(lazyQuery.load(document, variables));
```
goes:
graphql api only called once 

what make them different ?

**Comment by hahn-kev** [comment]
I'm also seeing this issue. I'm not sure why it's different for you, but it looks like when `variables` is changed a query gets fired off, and then another one again once load is finished calling. I believe it's because `variables` is being watched with `immediate` set to true and that might be the cause.

**Comment by ebisbe** [comment]
@Dykmi Why not use: 
```
const lazyQuery = useLazyQuery(document, variables);
lazyQuery.load();
```
if the document and variables don't change you can set them once.

**Comment by Csszabi98** [comment]
I did a little debugging related to this issue. It looks like it is a scheduler issue originating from vue.
Although vue does not guarantee that they will be called in order of definition. 

The real issue is a race condition between these two watchers:

https://github.com/vuejs/apollo/blob/v4/packages/vue-apollo-composable/src/useQuery.ts#L416
```
watch(variablesRef, (value, oldValue) => {
    const serialized = JSON.stringify(value)
    if (serialized !== currentVariablesSerialized) {
      currentVariables = value
      restart()
    }
    currentVariablesSerialized = serialized
  }, {
    deep: true,
    immediate: true,
  })
```

https://github.com/vuejs/apollo/blob/v4/packages/vue-apollo-composable/src/useQuery.ts#L529
```
watch(isEnabled, value => {
    if (value) {
      start()
    } else {
      stop()
    }
  }, {
    immediate: true,
  })
```

- The invocation order is totally dependent on vue
- If variablesRef watcher is invoked first query executes only once as expected
- If variablesRef watcher is invoked second query executes twice (isEnabled watcher has already initiated a query without variables set)

**Possible solution:**
- [Adjust flush timings](https://vuejs.org/guide/essentials/watchers.html#callback-flush-timing)

**Comment by hahn-kev** [comment]
wouldn't it be better to just make the race condition impossible by changing those 2 watchers into 1 watcher for both variables. Then just detect what changed and what you want to do based on the current state.

**Comment by adarshpk** [comment]
Like @Csszabi98 pointed out, The `load()` method sets `forceDisabled` to `false` hence triggering the isEnabled watcher, it also updates `query.variables.value` triggering the variablesRef watcher.

But since both watch are triggered from the same `load()` call, why would setTimeout make a difference?

**Comment by Akryum** [comment]
I can't reproduce on my end, could you post a minimal runnable repro ? Thanks

**Comment by Csszabi98** [comment]
I have managed to create a reproduction:

[vue-apollo-lazy-query-fired-twice.zip](https://github.com/vuejs/apollo/files/11689775/vue-apollo-lazy-query-fired-twice.zip)

The trigger to the issue is the following:
- Given that vue is busy (appending 100 new nodes in the example, may need to increase that number depending on your machine to get the bug)
- Watch A is invoked before Watch B, making the invocation of Watch A start a query without the variables
- Then Watch B restarts the query and executes it with the variables

As I have mentioned above Vue's incovation of the watchers does not guarantee an order of execution, they can be invoked in any order, hence this issue surfaces.

Snippets:

Watch A:
```javascript
watch(isEnabled, value => {
    if (value) {
      start()
    } else {
      stop()
    }
  }, {
    immediate: true,
  })
```

Watch B:
```javascript
watch(variablesRef, (value, oldValue) => {
    const serialized = JSON.stringify(value)
    if (serialized !== currentVariablesSerialized) {
      currentVariables = value
      restart()
    }
    currentVariablesSerialized = serialized
  }, {
    deep: true,
    immediate: true,
  })
```

Base Restart:

```javascript
 function baseRestart () {
   // Started will be true when Watch B runs first, hence the second query
   // In fact this would be the only query we want, but Watch A starts one prior to this
   // depending on the invocation order
    if (!started || restarting) return
    restarting = true
    // eslint-disable-next-line @typescript-eslint/no-floating-promises
    nextTick(() => {
      if (started) {
        stop()
        start()
      }
      restarting = false
    })
  }
  ```

**Comment by Csszabi98** [comment]
The issue is fixed with `"@vue/appollo-composable": "4.0.0-beta.5"`.
